functional-test/src/main/java/org/zanata/page/projects/projectsettings/ProjectLanguagesTab.java

```diff
@@ -111,37 +119,24 @@ public ProjectLanguagesTab enterSearchLanguage(String languageQuery) {
      * Add a language to the languages list.
      * Assumes that the search string has already been entered.
      *
-     * @param localeId language to select
+     * @param searchLocaleId language to select
      * @return new language settings, anticipating the language has been added.
      */
-    public ProjectLanguagesTab addLanguage(final String localeId) {
-        log.info("Click Add language on {}", localeId);
-        waitForAMoment().until(new Predicate<WebDriver>() {
+    public ProjectLanguagesTab addLanguage(final String searchLocaleId) {
+        log.info("Click Enable on {}", searchLocaleId);
+        String message = "can not find locale - " + searchLocaleId;
+        waitForAMoment().withMessage(message).until(new Predicate<WebDriver>() {
             @Override
             public boolean apply(WebDriver driver) {
-                List<WebElement> searchResults =
-                        WebElementUtil.getSearchAutocompleteResults(
-                                getDriver(),
-                                "settings-languages-form",
-                                "languageAutocomplete");
-
-                boolean clickedLocale = false;
-                for (WebElement searchResult : searchResults) {
-                    if (searchResult.getText().contains(localeId)) {
-                        searchResult.click();
-                        clickedLocale = true;
-                        break;
-                    }
-                }
-                return clickedLocale;
+                return LanguageList.toggleLanguageInList(
+                        getDriver().findElement(inactiveLocales),
+                        searchLocaleId);
             }
         });
-
-
         refreshPageUntil(this, new Predicate<WebDriver>() {
```

Comment:
This looks like the equivalent of pressing F5 repeatedly (going by the method name). Is that right?